### PR TITLE
Add provider FK to sources without db constraint

### PR DESCRIPTION
**Pull request by adberglund**
## Summary
This adds a foreign key relationship between the soruces and providers tables *without* creating a database constraint. Effectively we're creating a new column with no constraints, but Django sees it as a foreign key so we can actually join the tables using the ORM. 

Notice the absence of a `Foreign-key constraints:` section below
```
                   Table "public.api_sources"
       Column        |  Type   | Collation | Nullable | Default
---------------------+---------+-----------+----------+---------
 source_id           | integer |           | not null |
 source_uuid         | uuid    |           |          |
 name                | text    |           |          |
 auth_header         | text    |           |          |
 offset              | integer |           | not null |
 account_id          | text    |           |          |
 source_type         | text    |           | not null |
 authentication      | jsonb   |           | not null |
 billing_source      | jsonb   |           |          |
 koku_uuid           | text    |           |          |
 pending_delete      | boolean |           | not null |
 pending_update      | boolean |           | not null |
 out_of_order_delete | boolean |           | not null |
 status              | jsonb   |           |          |
 paused              | boolean |           | not null |
 provider_id         | uuid    |           |          |
Indexes:
    "api_sources_pkey" PRIMARY KEY, btree (source_id)
    "api_sources_koku_uuid_key" UNIQUE CONSTRAINT, btree (koku_uuid)
    "api_sources_source_uuid_key" UNIQUE CONSTRAINT, btree (source_uuid)
    "api_sources_provider_id_445890f5" btree (provider_id)
```

## Testing

Ran migration against local DB with test customers loaded. New provider_id column is populated. Deleted and created sources without issue. 

[Smoke Tests](https://ci.ext.devshift.net/job/project-koku-koku-pr-check/3272/)

### CODE REVIEW

File: koku/api/migrations/0051_sources_provider.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    operations = [
+        migrations.AddField(
+            model_name="sources",
+            name="provider",
+            field=models.ForeignKey(
+                db_constraint=False, null=True, on_delete=django.db.models.deletion.DO_NOTHING, to="api.provider"
+            ),
+        ),
+        migrations.RunSQL("UPDATE public.api_sources SET provider_id = koku_uuid::uuid"),
```
Comment (maskarb) [inline koku/api/migrations/0051_sources_provider.py:19]: Any idea what will happen here with a misconfigured source, e.g. one without a `koku_uuid`?

**Review by maskarb — COMMENTED** [review]
A question:

File: koku/api/migrations/0051_sources_provider.py (hunk 1 above)
Comment (adberglund) [reply koku/api/migrations/0051_sources_provider.py:19]:
```
koku=# select null::uuid;
 uuid
------

(1 row)
```

Comment (adberglund) [reply koku/api/migrations/0051_sources_provider.py:19]:
```
koku=# update api_sources set koku_uuid = null;
UPDATE 8
koku=# select koku_uuid from api_sources ;
 koku_uuid
-----------








(8 rows)

koku=# UPDATE public.api_sources SET provider_id = koku_uuid::uuid;
UPDATE 8
```

Comment (maskarb) [reply koku/api/migrations/0051_sources_provider.py:19]: Sweet, lgtm!

**Review by maskarb — DISMISSED** [review]

**Review by maskarb — APPROVED** [review]

**Pull request merged**